pkg/agent/controller/bgp/controller_test.go

```diff
@@ -2161,3 +2161,72 @@ func TestGetBGPPolicyInfo(t *testing.T) {
 		})
 	}
 }
+
+func TestGetBGPPeerStatus(t *testing.T) {
+	testCases := []struct {
+		name                  string
+		existingState         *bgpPolicyState
+		expectedCalls         func(mockBGPServer *bgptest.MockInterfaceMockRecorder)
+		expectedBgpPeerStatus []bgp.PeerStatus
+		expectedErr           string
+	}{
+		{
+			name:          "bgpPolicyState exists",
+			existingState: &bgpPolicyState{},
+			expectedCalls: func(mockBGPServer *bgptest.MockInterfaceMockRecorder) {
+				mockBGPServer.GetPeers(context.Background()).Return([]bgp.PeerStatus{
+					{
+						Address:      "192.168.77.200",
+						ASN:          65001,
+						SessionState: bgp.SessionEstablished,
+					},
+					{
+						Address:      "192.168.77.201",
+						ASN:          65002,
+						SessionState: bgp.SessionActive,
+					},
+				}, nil)
+			},
+			expectedBgpPeerStatus: []bgp.PeerStatus{
+				{
+					Address:      "192.168.77.200",
+					ASN:          65001,
+					SessionState: bgp.SessionEstablished,
+				},
+				{
+					Address:      "192.168.77.201",
+					ASN:          65002,
+					SessionState: bgp.SessionActive,
+				},
+			},
+		},
+		{
+			name:          "bgpPolicyState does not exist",
+			existingState: nil,
+			expectedErr:   ErrBGPPolicyNotFound.Error(),
```

Comment:
nit: `expectedErr` could be an `error` instead of a `string`, and you could use assertion `assert.ErrorIs` in the test case?